app/helpers/links_helper.rb

```diff
@@ -1,23 +1,146 @@
 module LinksHelper
-  def tracked_link_to(text, href, **kwargs)
-    govuk_link_to(text, href, **kwargs.deep_merge(data: {
+  def tracked_link_to(*args, **kwargs)
+    tracked_link_of_style(:govuk_link_to, *args, **kwargs)
+  end
+
+  def tracked_button_link_to(*args, **kwargs)
+    tracked_link_of_style(:govuk_button_link_to, *args, **kwargs)
+  end
+
+  def tracked_mail_to(*args, **kwargs)
+    tracked_link_of_style(:govuk_mail_to, *args, **kwargs)
+  end
+
+  def tracked_link_of_style(method_name, *args, **kwargs)
+    permitted_styles = %i[
+      govuk_button_link_to
+      govuk_link_to
+      govuk_mail_to
+    ]
+
+    raise ArgumentError, "Supports #{permitted_styles.to_sentence}" unless permitted_styles.include?(method_name)
+
+    send(method_name, *args, **kwargs.deep_merge(data: {
       controller: "tracked-link",
       action: %w[click auxclick contextmenu].map { |a| "#{a}->tracked-link#track" }.join(" "),
       "tracked-link-target": "link",
       "link-type": kwargs.delete(:link_type),
       "link-subject": kwargs.delete(:link_subject),
+      "tracked-link-text": kwargs.delete(:tracked_link_text),
     }))
   end
 
   def open_in_new_tab_link_to(text, href, **kwargs)
-    govuk_link_to("#{text} (opens in new tab)", href, target: "_blank", rel: "noreferrer noopener", **kwargs)
+    govuk_link_to(t("app.opens_in_new_tab", link_text: text), href, target: "_blank", rel: "noreferrer noopener", **kwargs)
   end
 
   def tracked_open_in_new_tab_link_to(text, href, **kwargs)
-    tracked_link_to("#{text} (opens in new tab)", href, target: "_blank", rel: "noreferrer noopener", **kwargs)
+    tracked_link_to(t("app.opens_in_new_tab", link_text: text), href, target: "_blank", rel: "noreferrer noopener", **kwargs)
+  end
+
+  def tracked_open_in_new_tab_button_link_to(text, href, **kwargs)
+    tracked_button_link_to(t("app.opens_in_new_tab", link_text: text), href, target: "_blank", rel: "noreferrer noopener", **kwargs)
   end
 
   def open_in_new_tab_button_link_to(text, href, **kwargs)
-    govuk_button_link_to("#{text} (opens in new tab)", href, target: "_blank", rel: "noopener", **kwargs)
+    govuk_button_link_to(t("app.opens_in_new_tab", link_text: text), href, target: "_blank", rel: "noopener", **kwargs)
+  end
+
+  def anon(value)
+    StringAnonymiser.new(value).to_s
+  end
+
+  def school_website_link(organisation, vacancy: nil, **kwargs)
+    tracked_open_in_new_tab_link_to(
+      t("schools.website_link_text", organisation_name: organisation.name),
+      organisation.url,
+      link_type: :school_website,
+      link_subject: anon(vacancy&.id),
```

Comment:
What does the & after vacancy do here? Probably my ignorance of Ruby but just to check!

Reply:
Because the vacancy is optional, we have to account for situations where it isn't present.  `vacancy&.id` means "call `id` on `vacancy` if `vacancy` is present, otherwise return nil`.

Reply:
No, because in this case we're passing in `vacancy_id` rather than the whole vacancy, due to where it's being used.  I think due to some changes we could now pass the whole vacancy in but it would be the same effect.

Reply:
Because the vacancy is optional, we have to account for situations where it isn't present.  `vacancy&.id` means "call `id` on `vacancy` if `vacancy` is present, otherwise return nil`.